contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -1574,20 +1587,26 @@ bool
 is_alter_server_stmt(GrantRoleStmt *stmt)
 {
 	/*
-	 * is alter server role statement, if one and the only one granted role is
+	 * is alter server role statement, if the granted role is
 	 * server role
 	 */
 
+	bool is_sysadmin = 0;
+
 	if (list_length(stmt->granted_roles) == 1)
 	{
 		RoleSpec   *spec = (RoleSpec *) linitial(stmt->granted_roles);
+		int     	rolename_len = strlen(spec->rolename);
+
+		if (rolename_len == 8 && strncmp(spec->rolename, "sysadmin", 8) == 0)
+			is_sysadmin = true;
 
-		if (strcmp(spec->rolename, "sysadmin") == 0)	/* only supported server
-														 * role */
+		/* only supported server roles */
+		if (is_sysadmin || (rolename_len == 13 && strncmp(spec->rolename, "securityadmin", 13) == 0))
```

Comment:
Keeping in mind that the alter role stmt also goes through this path, we should keep this check with strncmp instead strcmp since the rolename can be something different.

Reply:
```
if (strcmp(spec->rolename, BABELFISH_SYSADMIN) == 0 || strcmp(spec->rolename, BABELFISH_SECURITYADMIN) == 0)
```
I meant using this check instead. Just to improve readability.

Reply:
But strncmp is more preferrable over strcmp since strcmp is more vulnerable compared to strncmp.